### PR TITLE
Build on JDK 8, 11, 14

### PR DESCRIPTION
**Pull request by dblock**
Signed-off-by: dblock <dblock@dblock.org>

### Description

Added CI for JDK 8, 11 and 14.

### Issues Resolved

Closes https://github.com/opensearch-project/asynchronous-search/issues/66.
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by reta** [comment]
@dblock same for JDK-17, we need Gradle 7 but `Welcome to Gradle 6.7.1!` :-)

**Review by reta — DISMISSED** [review]

File: build.gradle [hunk 1]
```diff
@@ -71,7 +71,7 @@ ext {
     noticeFile = rootProject.file('NOTICE.txt')
 }
 
-sourceCompatibility = 1.9
+sourceCompatibility = targetCompatibility = "1.8"
```
Comment (reta) [inline build.gradle:74]: Super minor but maybe more conventional: 
```
sourceCompatibility = 1.8
targetCompatibility = 1.8
```

Comment (dblock) [reply build.gradle:74]: We kinda do both, but yes.

**Review by Bukhtawar — APPROVED** [review]
Thanks @dblock for picking this up

**Review by dhruv16dhr — APPROVED** [review]

**Pull request merged**